### ISSUE TITLE
Bug in qwen-vl-finetune/qwenvl/data/data_processor.py/preprocess_qwen_visual: Bad things happen when token 77091(assistent) appears in some position

### ISSUE THREAD

**Issue by Singularity0104** [QwenLM/Qwen3-VL#2010]
I notice that the `preprocess_qwen_visual` in `data_processor.py` uses token 77091 to define labels:
``` python
    messages = _build_messages(source, base_path)

    full_result = processor.apply_chat_template(
        messages, tokenize=True, return_dict=True, return_tensors="pt"
    )

    input_ids = full_result["input_ids"]
    if isinstance(input_ids, list):
        input_ids = torch.tensor(input_ids).unsqueeze(0)

    labels = torch.full_like(input_ids, IGNORE_INDEX)

    input_ids_flat = input_ids[0].tolist()
    L = len(input_ids_flat)
    pos = 0
    while pos < L:
        if input_ids_flat[pos] == 77091:
            ans_start = pos + 2
            ans_end = ans_start
            while ans_end < L and input_ids_flat[ans_end] != 151645:
                ans_end += 1
            if ans_end < L:
                labels[0, ans_start : ans_end + 2] = input_ids[
                    0, ans_start : ans_end + 2
                ]
                pos = ans_end
        pos += 1

    full_result["labels"] = labels
    full_result["input_ids"] = input_ids
```
However, if token 77091(*assistant* without leading spaces) appears in other position, it will identified as the start of the label and participate in the loss calculation. For example
``` python
from transformers import AutoProcessor
import torch

IGNORE_INDEX = -100

model_path = "Qwen3-VL-8B-Instruct"

processor = AutoProcessor.from_pretrained(model_path)

# print(processor)

message = [
    {"role": "user","content": [{"type": "text", "text": "assistant: false assistant content"},]},
    {"role": "assistant","content": [{"type": "text", "text": "assistant: true assistant content"},],
}]

text = processor.apply_chat_template(
    message, tokenize=False, return_dict=True, return_tensors="pt"
)
print('==================text==================')
print(text)

full_result = processor.apply_chat_template(
    message, tokenize=True, return_dict=True, return_tensors="pt"
)
print('==================ids===================')
print(list(full_result['input_ids'][0].cpu().numpy()))

input_ids = full_result["input_ids"]
if isinstance(input_ids, list):
    input_ids = torch.tensor(input_ids).unsqueeze(0)

labels = torch.full_like(input_ids, IGNORE_INDEX)

input_ids_flat = input_ids[0].tolist()
L = len(input_ids_flat)
pos = 0
while pos < L:
    if input_ids_flat[pos] == 77091:
        ans_start = pos + 2
        ans_end = ans_start
        while ans_end < L and input_ids_flat[ans_end] != 151645:
            ans_end += 1
        if ans_end < L:
            labels[0, ans_start : ans_end + 2] = input_ids[
                0, ans_start : ans_end + 2
            ]
            pos = ans_end
    pos += 1

print('=================labels=================')
print(list(labels[0].cpu().numpy()))
```
will give the results:

<img width="920" height="149" alt="Image" src="https://github.com/user-attachments/assets/009c3cdf-07a0-4166-bc6a-fe12ed20dc28" />

Some of the token ids in the user content part is incorrectly exposed to the labels.

**Comment by JJJYmmm** [comment]
thank you for pointing out this bug.  We should add a extra check for `<|im_start|>=151644` before the assistant token, e.g. `if input_ids_flat[pos] == 77091 and input_ids_flat[pos - 1] == 151644:`(pos should start from 1 since im_start is also at pos 0 when fine-tuning). Could you open a PR for that?

**Comment by Singularity0104** [comment]
@JJJYmmm Sure. #2017
